app/src/main/java/com/codesoom/assignment/controller/TodoHttpController.java

```diff
@@ -0,0 +1,54 @@
+package com.codesoom.assignment.controller;
+
+import com.codesoom.assignment.models.Todo;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class TodoHttpController {
+    private List<Todo> todoList;
+    private Integer id;
+    public TodoHttpController() {
+        this.todoList = new ArrayList<>();
+        this.id = 0;
+    }
+
+    public List<Todo> getTodos() {
+        return this.todoList;
+    }
+
+    public Todo getTodos(String id) {
+        return this.todoList.stream()
+                .filter(t -> t.getId().equals(Integer.parseInt(id)))
+                .findFirst()
+                .get();
```

Comment:
이렇게 하면 어떨까요? 어떤 차이점이 있나요?

```suggestion
        final int idNumber = Integer.parseInt(id);
        return this.todoList.stream()
                .filter(t -> t.getId().equals(idNumber))
                .findFirst()
                .get();
```

Reply:
@johngrib 
잘 모르겠습니다. 차이점이 뭔가요?

Reply:
다음 코드를 실행해 보세요.

```java
    private int printAndParseInt(String number) {
        System.out.println(number);
        return Integer.parseInt(number);
    }

    @Test
    void test1() {
        final List<Integer> idList = List.of(1, 2, 3, 4, 5, 6);
        final int idNumber = printAndParseInt("6");
        int result = idList.stream()
                .filter(t -> t == idNumber)
                .findFirst()
                .get();
     Assertions.assertTrue(true);
    }

    @Test
    void test2() {
        final List<Integer> idList = List.of(1, 2, 3, 4, 5, 6);
        final String idNumber = "6";
        int result = idList.stream()
                .filter(t -> t == printAndParseInt(idNumber))
                .findFirst()
                .get();
        Assertions.assertTrue(true);
    }
```